### DISCUSSION TITLE
multiple authorizations [v5]

### DISCUSSION THREAD

**Discussion by sasha172017** [adonisjs/core#1785] [Help]
Can i`ll make multiple auth in adonis? i need tables user, staff and models user, staff and for them should be own registration and auth. Can i later receive user and staff like this - ctx.auth.user, ctx.auth.staff. And actually i want than make one model api_tokens for this auth. How can i do it? thank you=)

**Comment by WeyIin** [comment]
Just have a flag in the users table to represent whether it is a normal user or staff. 1 authentication, 1 table, 1 model :)  
Don't overcomplicate things.

id | username | staff
------------ | ------------- | -------------
1 | Staff 1 | true
2 | User 1 | false

**Reply by sasha172017** [reply]
It isn't good decision... I need two table, two model, two auth, it architecture is possibility in Laravel

**Comment by mcordoba** [comment]
I did it in v4.1 having several auths.

```javascript
const authenticator = auth.authenticator('jwt_user')
      result = await authenticator
        .query(extraConditions)
        .attempt(data.email, data.password, true)
```

Then, in config, you must add:

```javascript
jwt_user: {
    serializer: 'lucid',
    model: 'App/Models/User',
    scheme: 'jwt',
    uid: 'email',
    password: 'password',
    options: {
      secret: Env.get('APP_KEY'),
      audience: 'user'
    }
  },

  jwt_staff: {
    serializer: 'lucid',
    model: 'App/Models/Staff',
    scheme: 'jwt',
    uid: 'email',
    password: 'password',
    options: {
      secret: Env.get('APP_KEY'),
      audience: 'staff'
    }
  },
```

So in v5 should be similar.

**Reply by sasha172017** [reply]
I not wanna to use jwt, i'll want to use table api tokens, for create/save tokens
